.github/workflows/review-code.yaml

```diff
@@ -40,17 +40,21 @@ jobs:
       # Step 5: Send PR Diff for Each File Separately
       - name: Send PR Diff for Each File Separately
         run: |
-          diff_content=$(cat diff.txt | jq -Rsa .)
           # Loop through each file path and send it in a separate request
           while IFS= read -r file; do
             echo "Processing file: $file"
-            file_content=$(git show "${{ github.event.pull_request.head.sha }}:$file" | jq -Rsa .)  # Get the full file content
+
+            # Get the specific diff for this file
+            file_diff=$(git diff -U3 origin/${{ github.event.pull_request.base.ref }}...origin/${{ github.event.pull_request.head.ref }} -- "$file" | jq -Rsa .)
```

Comment:
This line violates the Single Responsibility Principle (SRP) as it combines multiple responsibilities: fetching the git diff and processing it with jq. (Severity: medium)